src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -77,6 +86,36 @@ private static ResultType extractResultType(Statement statement) throws SQLExcep
     }
   }
 
+  // Split statements by ';' delimiter, but ignore anything that is nested with '' or "".
```

Comment:
How about ';' inside comments? Should we first remove all comments, is this already done at a higher layer? Can you explain?

Can you add a test for `;` inside comments?

Reply:
I added a check to remove comments.